lib/pending-processor.js

```diff
@@ -14,6 +17,13 @@ var SYNC_UPDATE_TYPES = {
 };
 
 function saveUpdate(datasetId, pendingChange, type, msg, callback) {
+  if (type !== SYNC_UPDATE_TYPES.APPLIED && pendingChange.tries < retryLimit) {
+    //if the pending change can not be applied, but it hasn't reach the retry limit, 
+    //let's not do anything and just leave it in the queue by returning an error. 
+    //it will be tried again.
+    //evetually when pendingChange.tries == retryLimit, the right status will be saved.
+    return callback(msg || "retry"); 
```

Comment:
Could it be `callback(new Error(msg || "retry"));` to use the same type as in other callbacks

Reply:
yes